packages/components/src/components/viewers/ImageView.tsx

```diff
@@ -8,54 +11,64 @@ enum LoadingState {
 }
 
 interface ViewerProps {
-  file: string
-  setError: (error: Error) => void
+  parsedKey: UrlKey | FileKey
```

Comment:
not sure that `parsedKey` is the name I would go for here. Maybe `content`? In the other repo I moved toward ViewerProps accepting:

```
interface ViewerProps {
  content: AsyncBufferFrom
  ...
}
export type AsyncBufferFrom = FileContent<void> | AsyncBufferFromFile
```

So basically solving the same problem but slightly different names. Ultimately what's important is that it is an object that defines how to construct an AsyncBuffer object. It needs to be a serializable object so that we can send it to the webworker. And the two cases are either: from a url or from a dropped file.

In the case of a UrlKey: it should also include a byteLength property. Why? Otherwise we might need to make a HEAD request every time we want to construct an AsyncBuffer.

This is ok for now for merging the PR. We can evolve this over time. But I think both repos have not found exactly the right abstraction here.

Reply:
exactly! I couldn't find a good name, and `key` is a reserved word in React so we cannot use it to pass the data as `key={key}`.
Note that my intent is also to do the URL/path parsing once, and avoid doing it again (will be maybe more important when we will include the parsing of HF URLs, since the different parts of the URL have different meanings, see https://github.com/severo/hyparam--space/blob/main/src/lib/huggingface.ts).

Maybe the abstraction is somewhat like a filesystem (as in https://filesystem-spec.readthedocs.io): being able to handle and navigate through files stored locally, remotely on a standard HTTP hosting, on S3, on HF, etc. using the same code.